mlinspect/monkeypatching/_patch_pandas.py

```diff
@@ -114,6 +114,39 @@ def execute_inspections(op_id, caller_filename, lineno, optional_code_reference,
 
         return execute_patched_func(original, execute_inspections, self, *args, **kwargs)
 
+    @gorilla.name('sample')
+    @gorilla.settings(allow_hit=True)
+    def patched_sample(self, *args, **kwargs):
+        """ Patch for ('pandas.core.frame', 'sample') """
+        original = gorilla.get_original_attribute(pandas.DataFrame, 'sample')
+
+        def execute_inspections(op_id, caller_filename, lineno, optional_code_reference, optional_source_code):
+            """ Execute inspections, add DAG node """
+            function_info = FunctionInfo('pandas.core.frame', 'sample')
+
+            input_info = get_input_info(self, caller_filename, lineno, function_info, optional_code_reference,
+                                        optional_source_code)
+            operator_context = OperatorContext(OperatorType.SELECTION, function_info)
+            input_infos = PandasBackend.before_call(operator_context, [input_info.annotated_dfobject])
+            # No input_infos copy needed because it's only a selection and the rows not being removed don't change
+            result = original(input_infos[0].result_data, *args[1:], **kwargs)
+            if result is None:
+                raise NotImplementedError("TODO: Support inplace sample")
```

Comment:
I checked the [pandas documentation](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.sample.html) just now and it seems like there is no `inplace` argument. I think we can remove this if-branch then. And I should probably fix this issue soon in the place where you copied it from :D